ide/svg/src/org/netbeans/modules/svg/SVGViewerElement.java

```diff
@@ -0,0 +1,382 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.svg;
+
+import com.github.weisj.jsvg.SVGDocument;
+import com.github.weisj.jsvg.geometry.size.FloatSize;
+import com.github.weisj.jsvg.parser.SVGLoader;
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.swing.Action;
+import javax.swing.JComponent;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JToolBar;
+import javax.swing.SwingConstants;
+import org.netbeans.core.spi.multiview.CloseOperationState;
+import org.netbeans.core.spi.multiview.MultiViewElement;
+import org.netbeans.core.spi.multiview.MultiViewElementCallback;
+import org.netbeans.modules.svg.toolbar.SVGViewerToolbar;
+import org.openide.awt.UndoRedo;
+import org.openide.filesystems.FileChangeAdapter;
+import org.openide.filesystems.FileChangeListener;
+import org.openide.filesystems.FileEvent;
+import org.openide.filesystems.FileObject;
+import org.openide.util.Lookup;
+import org.openide.util.NbBundle;
+import org.openide.util.NbBundle.Messages;
+import org.openide.util.lookup.Lookups;
+import org.openide.util.lookup.ProxyLookup;
+import org.openide.windows.TopComponent;
+
+/**
+ *
+ * @author Christian Lenz
+ */
+@MultiViewElement.Registration(
+    displayName = "#LBL_SVGViewer",
+    iconBase = "org/netbeans/modules/svg/resources/svgLogo.png",
+    mimeType = "image/svg+xml",
+    persistenceType = TopComponent.PERSISTENCE_NEVER,
+    preferredID = "SVGViewer",
+    position = 1100
+)
+@Messages("LBL_SVGViewer=Preview")
+public class SVGViewerElement implements MultiViewElement {
+
+    static {
+        // JSVG loader/parser is logging with SEVERE level which would land in the NB exception reporter
+        Logger.getLogger("com.github.weisj.jsvg").setLevel(Level.OFF);
+    }
```

Comment:
thought about this a bit more: it might be better to set this via the [config file](https://github.com/apache/netbeans/blob/daa8132538e9343d013cbac3c20cff859aefa5f4/nb/ide.launcher/netbeans.conf#L59). Since turning the logger on again might help @eirikbakke to debug icon loading.

I might be looking at the wrong places but I can't find any logging configs anywhere - a bit surprising that NB didn't have to adjust log levels of third party libs before. cc @neilcsmith-net

Reply:
Perhaps set it to level Level.INFO instead? That way the error ends up on the console but does not trigger the exception reporter in the user interface.

Reply:
i think INFO is the default. Keep in mind that this setting is the cutoff for the log handler and does not influence the message log levels. The lib logs with SEVERE, unfortunately the next higher level is OFF.

Reply:
> Keep in mind that this setting is the cutoff for the log handler and does not influence the message log levels. 

Oh... you're right. Idea: Turn the logging off only temporarily while the file is being loaded for SVG Preview purposes, but then turn it back to its previous value in a "finally" block right after.

Reply:
wouldn't recommend doing that for two reasons: threading and it being brittle since the places where the lib logs with SEVERE could change. Turning logging off for that package is just configuration - thats why I thought to move it to the NB config.
 
the spec is that it returns null if it fails - we have to live with it i guess. For debugging reasons we can switch it on again.

Reply:
OK, I'm fine with turning logging off for JSVG then, by configuration or otherwise.

Perhaps eventually we could request from @weisJ that JSVG reduces the log level from SEVERE to INFO for problems that are caused by invalid SVG inputs, rather than true unexpected conditions?